client/rpc/src/eth.rs

```diff
@@ -1093,6 +1093,7 @@ where
 			self.client.as_ref(),
 			self.backend.as_ref(),
 			hash,
+			false,
```

Comment:
@nanocryk I supposed that getTransactionByHash is returning the transaction even if not in canonical, is that correct ?

Reply:
I think we can return only canon transactions here as well.